magento/pwa-studio
# My Account - Forgot Password

## Description

Implemented forgot password UX and workflow for the new my account feature.

For this PR I had to add a new route `/customer/account/createPassword` which provides the UX to reset the user's password.

When the user requests a password reset, the backend sends an email with a button to take the user to a new tab. The link that the backend includes in the email, uses the MAGENTO_BACKEND_URL as the domain. Since all the backends don't have the code to handle the new route `/customer/account/createPassword`, @dpatil-magento helped me deploy the new code on this server: https://staging-vdt2zeq-c5v7sxvquxwl4.eu-4.magentosite.cloud/. If someone wants to test this PR locally, please use the above backend URL instead.

Also if you wanna test this PR on a deployed instance please use #2621. @dpatil-magento made sure the backend for that PR is set to https://staging-vdt2zeq-c5v7sxvquxwl4.eu-4.magentosite.cloud/ which has the code to handle the new route.

Another way to test this is to paste the query string part from the email link next to the deployed app URL, for instance: `https://pr-2619.pwa-venia.com/customer/account/createPassword/` + `?email=gooseton%40gmail.com&token=*******`.

Sorry if this is confusing.

**Note: To be merged only after #2590** 

## Related Issue
Closes [PWA-77](https://jira.corp.magento.com/browse/PWA-77)

## Acceptance 
Users should be able to reset their password.

### Verification Stakeholders
@soumya-ashok 
@schensley 
@jimbo 

### Specification
This PR adds a new route `customer/account/createPassword` which takes 2 query params `?email=EMAIL_ID&token=RESET_TOKEN`. 

### Verification Steps
1. Go to the deployed app and open the sign-in dropdown. Click on the forgot password link.
2. Enter the email id and submit.
3. Check for the reset email. Click the reset button in the email.
4. Reset UI should open up in the new tab.
5. Enter new password and click submit.

## Screenshots / Screen Captures (if appropriate)

Demo link: https://jira.corp.magento.com/secure/attachment/367170/Forgot%20password%20demo.mp4

![image](https://user-images.githubusercontent.com/35203638/90071290-ead3d380-dcba-11ea-9842-b6046708d6f0.png)

![image](https://user-images.githubusercontent.com/35203638/90667362-47cc0e00-e214-11ea-9fa3-49a7e399966c.png)

![image](https://user-images.githubusercontent.com/35203638/90567488-d8044780-e16f-11ea-89ac-cefbde261d0a.png)

![image](https://user-images.githubusercontent.com/35203638/90074670-528d1d00-dcc1-11ea-8880-62daf4014ab5.png)

![image](https://user-images.githubusercontent.com/35203638/90676997-5e796180-e222-11ea-8e7d-22eb9c544d40.png)

![image](https://user-images.githubusercontent.com/35203638/90677028-69cc8d00-e222-11ea-84cc-31202c6bc6de.png)

![image](https://user-images.githubusercontent.com/35203638/90677100-84066b00-e222-11ea-9ffc-64d84c7bdf47.png)

![image](https://user-images.githubusercontent.com/35203638/90677168-94b6e100-e222-11ea-89b4-322d709e5bf0.png)

![image](https://user-images.githubusercontent.com/35203638/90567685-287ba500-e170-11ea-874b-6982bdade693.png)

## Checklist

* I have updated the documentation accordingly, if necessary.
